packages/babel-traverse/src/scope/index.ts

```diff
@@ -315,11 +316,27 @@ const collectorVisitor: Visitor<CollectVisitorState> = {
     }
   },
 
-  BlockScoped(path) {
+  BlockScoped(path, state) {
     let scope = path.scope;
     if (scope.path === path) scope = scope.parent;
 
     const parent = scope.getBlockParent();
+
+    if (
+      path.isFunctionDeclaration({ async: false, generator: false }) &&
```

Comment:
nit: I prefer to move `sloppyFunctionDeclarationStack` manipulation to the `Function` visitor and then split `BlockScoped` visitor, as the current logic is implicitly determined by the execution order of `Function` and `BlockScoped` visitor.

Reply:
Yes sure, I'll be happy to make these changes. Give me a day or two please.